src/lib.rs

```diff
@@ -351,3 +374,236 @@ pub async fn delete_one(
+impl From<Order> for PropertyOrderDirection {
+    fn from(item: Order) -> Self {
+        match item {
+            Order::Ascending => PropertyOrderDirection::Ascending,
+            Order::Descending => PropertyOrderDirection::Descending,
+        }
+    }
+}
+
+pub struct DatastorersPropertyFilterItem {
+    pub value: DatastoreValue,
+    pub operator: Operator,
+    pub property: String,
+}
+
+impl From<DatastorersPropertyFilterItem> for Filter {
+    fn from(filter_item: DatastorersPropertyFilterItem) -> Self {
+        let operator = filter_item.operator.into();
+        let filter = PropertyFilter {
+            property: Some(PropertyReference {
+                name: Some(String::from(filter_item.property)),
+            }),
+            value: Some(filter_item.value.into()),
+            op: Some(operator),
+        };
+        Filter {
+            property_filter: Some(filter),
+            composite_filter: None,
+        }
+    }
+}
+
+pub struct DatastorersPropertyFilter {
+    pub filter_items: Vec<DatastorersPropertyFilterItem>,
+}
+
+impl Default for DatastorersPropertyFilter {
+    fn default() -> Self {
+        DatastorersPropertyFilter {
+            filter_items: Vec::new(),
+        }
+    }
+}
+
+impl DatastorersPropertyFilter {
+
+    pub fn push(&mut self, filter_prop: String, operator: Operator, value: DatastoreValue) {
+        self.filter_items.push(DatastorersPropertyFilterItem {
+            value,
+            operator,
+            property: filter_prop
+        });
+    }
+}
+
+impl TryFrom<DatastorersPropertyFilter> for Filter {
+    type Error = DatastorersError;
+
+    fn try_from(mut val: DatastorersPropertyFilter) -> Result<Self, Self::Error> {
+        match val.filter_items.len() {
+            0 => Err(DatastoreClientError::NoFilterProps)?,
+            1 => {
+                let filter_item = val.filter_items.remove(0);
+                Ok(filter_item.into())
+            },
+            _ => {
+                let composite_filter = CompositeFilter {
+                    op: Some(CompositeFilterOp::And), // Only allowed value
+                    filters: Some(val.filter_items
+                        .into_iter()
+                        .map(|filter_item| filter_item.into())
+                        .collect()
+                    ),
+                };
+                Ok(Filter {
+                    property_filter: None,
+                    composite_filter: Some(composite_filter),
+                })
+            }
+        }        
+    }
+}
+
+pub struct DatastorersQuery<E> {
+    entity: PhantomData<E>,
+    filter: Option<DatastorersPropertyFilter>,
+    limit: Option<i32>,
+    order: Vec<PropertyOrder>,
+}
+
+
+impl<E> Default for DatastorersQuery<E> {
+    fn default() -> Self {
+        DatastorersQuery {
+            entity: PhantomData,
+            filter: None,
+            limit: None,
+            order: Vec::new(),
+        }
+    }
+}
+
+impl<E> DatastorersQuery<E> 
+where
```

Comment:
if you move the bounds to the struct the rules are "tighter"?

Reply:
Hmm, but how?

I removed these bounds and instead hange the struct to:

`pub struct DatastorersQuery<E> where
E: Kind + Pagable + DatastorersQueryable<E> + TryFrom<DatastoreEntity, Error = DatastorersError>
{`

I get 2000 errors and does not really know where to start fixing, do not really get it...

Reply:
I think you need the bounds on both...